packages/core/test/unit/utils/parse-common-connection-url-options.test.ts

```diff
@@ -0,0 +1,140 @@
+import { parseCommonConnectionUrlOptions } from '@sequelize/core/_non-semver-use-at-your-own-risk_/utils/connection-options.js';
+import { expect } from 'chai';
+
+describe('parseCommonConnectionUrlOptions', () => {
+  it('maps url parts to specified options', () => {
+    const options = parseCommonConnectionUrlOptions<any>({
+      stringSearchParams: ['search1', 'search2'],
+      booleanSearchParams: ['search3'],
+      numberSearchParams: ['search4'],
+      allowedProtocols: ['mariadb:'],
+      hostname: 'theHost',
+      password: 'thePassword',
+      pathname: 'theDb',
+      port: 'thePort',
+      url: 'mariadb://a:b@c:1234/d?search1=1&search2=2&search3=true&search4=4',
+      username: 'theUser',
+    });
+
+    expect(options).to.deep.eq({
+      theDb: 'd',
+      theHost: 'c',
+      thePassword: 'b',
+      thePort: 1234,
+      theUser: 'a',
+      search1: '1',
+      search2: '2',
+      search3: true,
+      search4: 4,
+    });
+  });
+
+  it('throws when specifying an unknown search param', () => {
+    expect(() =>
+      parseCommonConnectionUrlOptions<any>({
+        url: 'mariadb://localhost?connectTimeout=1000',
+        allowedProtocols: ['mariadb:'],
```

Comment:
This feels counter intuitive, I'd prefer if the `:` is added by sequelize instead of explicitly given.

Reply:
Done

Reply:
I see now, that's an interesting choice to me. But your solution is elegant, I like it

Reply:
I looked into why they did that but as I haven't found any reason, I opted to just remove it as you proposed